_posts/2024-12-17-vscode-osd-setup.markdown

```diff
@@ -0,0 +1,282 @@
+
+After performing these steps, you should see the following **Test Explorer** (🧪) icon in the **Extensions** sidebar. 
+
+Select this icon to view all detected test files.
+
+## Configuring Prettier and ESLint integration
+
+The OSD repo includes [Husky](https://typicode.github.io/husky/), a precommit hook that runs scripts before commits are made (scripts may include linting or running unit tests). However, rerunning the linter and recommitting your changes may still be time-consuming. VS Code provides the ability to lint on save but OSD has specific linting rules that may not work out of the box.
+
+You can use the [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) VS Code plugins to lint files automatically on save. Note that some rules that cannot be fixed automatically will still require manual corrections. The Prettier plugin is configured using the `.prettierrc` config file and will execute the Prettier rules in VS Code. ESLint is configured using the workspace's ESLInt configuration to enforce its rules. The linting integration is shown in the following image.
+
+![linting-integration](/assets/media/blog-images/2024-12-17-vscode-osd-setup/linting-integration.gif)
+
+### Prerequisites
+
+Before you start, configure linting on save by pressing `Cmd` (or `Ctrl`) + `Shift` + `P` and selecting **Open User Settings (JSON)** so you can save this setting across all your projects:
+
+```json
+{
+    "editor.formatOnSave": true
+}
+```
+
+### Setup
+
+1. Install the Prettier and ESLint extensions.
+2. Press `Cmd` (or `Ctrl`) + `Shift` + `P` and select **Open Workspace Settings (JSON)**.
+3. Add in the following settings to the `settings.json` file:
+
+    ```json
+    {
+      "prettier.configPath": ".prettierrc",
+      "editor.defaultFormatter": "esbenp.prettier-vscode",
+      "eslint.autoFixOnSave": true,
+      "editor.codeActionsOnSave": {
+        "source.fixAll.eslint": true
+      },
+    }
+    ```
+**Note**: If you already have some settings in the `editor.codeActionsOnSave` setting, append `"source.fixAll.eslint": true` to the existing settings.
+
+Now, when you save a file, the file is linted automatically.
+
+## Configuring OSD server run tasks
+
+Starting the OSD server typically requires running commands in two terminals:
+
+* In the first terminal, run `yarn opensearch snapshot`.
+* After several seconds, in the second terminal, run `yarn run start --no-base-path`.
+
+This approach works for many use cases but doesn't allow you to debug server-side changes. To simplify this process, you can set up a series of launch tasks, turning server startup into a one-click operation:
+
+1. Start the OpenSearch server.
+2. Start the Dashboards server. Starting the Dashboards server requires a wait period for the server to boot up; this can be accomplished with a VS Code task to wait for a period of time.
```

Comment:
```suggestion
2. Start the Dashboards server. Starting the Dashboards server requires a wait period while the server boots up; this can be accomplished with a VS Code task that specifies to wait for a certain period of time.
```